IHP/ModelSupport.hs

```diff
@@ -566,21 +566,46 @@ class
     --
     columnNames :: [ByteString]
 
-    -- | Returns WHERE conditions to match an entity by it's primary key
+    -- | Returns the list of column names, that are contained in the primary key for a given model
+    --
+    -- __Example:__
+    --
+    -- >>> primaryKeyColumnNames @User
+    -- ["id"]
+    --
+    -- >>> primaryKeyColumnNames @PostTagging
+    -- ["post_id", "tag_id"]
+    --
+    primaryKeyColumnNames :: [ByteString]
+
+    -- | Returns WHERE conditions to match an entity by it's primary key, given the entities id
     --
     -- For tables with a simple primary key this returns a tuple with the id:
     --
-    -- >>> primaryKeyCondition project
+    -- >>> primaryKeyConditionForId project.id
     -- [("id", "d619f3cf-f355-4614-8a4c-e9ea4f301e39")]
     --
     -- If the table has a composite primary key, this returns multiple elements:
     --
-    -- >>> primaryKeyCondition postTag
+    -- >>> primaryKeyConditionForId postTag.id
     -- [("post_id", "0ace9270-568f-4188-b237-3789aa520588"), ("tag_id", "0b58fdf5-4bbb-4e57-a5b7-aa1c57148e1c")]
-    --
-    primaryKeyCondition :: record -> [(Text, PG.Action)]
-    default primaryKeyCondition :: forall id. (HasField "id" record id, ToField id) => record -> [(Text, PG.Action)]
-    primaryKeyCondition record = [("id", toField record.id)]
+    primaryKeyConditionForId :: Id record -> [(Text, PG.Action)]
+    default primaryKeyConditionForId :: (ToField (Id record)) => Id record -> [(Text, PG.Action)]
+    primaryKeyConditionForId id = [("id", toField id)]
```

Comment:
This is technically not correct, I am assuming here again, that the primary key column is actually called "id". However i don't really see a way around this, if we wanna provide a default implementation. 
Is there already constraint that checks whether "id" is actually a real field within the table?
We could of course generate such a constraint through SchemaCompiler, however that kind of defeats the purpose of the default implementation, because if a user is using SchemaCompiler, they aren't using the default implementation.
I think the cleanest would be to just remove the default implementation, as in most cases it is generated by the SchemaCompiler anyways... Are there any reasons to keep it?

Reply:
> Are there any reasons to keep it?

Not really, it can be removed without breaking anything in practise